src/main/kotlin/com/bardsoftware/backend/fleet/rmarkdown/Subscriber.kt

```diff
@@ -154,10 +166,10 @@ class SubscribeManager(subscriptionId: String,
             subscriber.stopAsync()
         }
     }
+}
 
-    fun pushMessage(taskId: String, rootFileName: String, zipBytes: ByteString): File {
-        return this.receiver.unzipCompileTask(taskId, rootFileName, zipBytes)
-    }
+fun subscribe(subscription: String, receiver: CompilerFleetMessageReceiver) {
+    SubscribeManager(subscription, receiver).subscribe()
```

Comment:
Do you really need to create a new instance of class? Why don't you do all the stuff in a function?